lib/static/new-ui/features/suites/components/BrowsersSelect/index.tsx

```diff
@@ -0,0 +1,182 @@
+import {Globe} from '@gravity-ui/icons';
+import {Button, Flex, Select, SelectRenderControlProps, SelectRenderOption} from '@gravity-ui/uikit';
+import React, {useState, useEffect, ReactNode} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import * as actions from '@/static/modules/actions';
+import {BrowserIcon} from '@/static/new-ui/features/suites/components/BrowsersSelect/BrowserIcon';
+import {State} from '@/static/new-ui/types/store';
+import {BrowserItem} from '@/types';
+import styles from './index.module.css';
+
+// In the onUpdate callback we only have access to array of selected strings. That's why we need to serialize
+// id/version in string. Encoding to avoid errors if id/version contains delimiter.
+// The other approach would be to use mapping, but in practice it makes things even more complex.
+const DELIMITER = '/';
+
+const serializeBrowserData = (id: string, version: string): string =>
+    `${encodeURIComponent(id)}${DELIMITER}${encodeURIComponent(version)}`;
+
+const deserializeBrowserData = (data: string): {id: string; version: string} => {
+    const [idEncoded, versionEncoded] = data.split(DELIMITER);
+    return {id: decodeURIComponent(idEncoded), version: decodeURIComponent(versionEncoded)};
+};
+
+interface BrowsersSelectProps {
+    browsers: BrowserItem[];
+    filteredBrowsers: BrowserItem[];
+    actions: typeof actions;
+}
+
+function BrowsersSelectInternal({browsers, filteredBrowsers, actions}: BrowsersSelectProps): ReactNode {
+    const [selectedBrowsers, setSelectedBrowsers] = useState<BrowserItem[]>([]);
+
+    useEffect(() => {
+        setSelectedBrowsers(filteredBrowsers);
+    }, [browsers, filteredBrowsers]);
+
+    const renderFilter = (): React.JSX.Element => {
+        return (
+            <div className={styles['browserlist__filter']}>
+                <Button onClick={(): void => setSelectedBrowsers(browsers)} width='max'>
+                    Select All
+                </Button>
+            </div>
+        );
+    };
+
+    const onUpdate = (values: string[]): void => {
+        const selectedItems: BrowserItem[] = [];
+
+        values.forEach(encodedBrowserData => {
+            const {id, version} = deserializeBrowserData(encodedBrowserData);
+            const existingBrowser = selectedItems.find(browser => browser.id === id);
+
+            if (existingBrowser) {
```

Comment:
Not sure what you're implying. Here we generate new state based on the new array of encoded values. selectedItems here is initally an empty array. At first no browsers exist there. We check if we've already added such browser before, then we add just a version. Otherwise we add the whole new browser.